src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/matrix_nms/matrix_nms_kernel_ref.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "matrix_nms_kernel_ref.h"
+
+#include <vector>
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey MatrixNmsKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::INT32);
+
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+
+    k.EnableDifferentTypes();
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    return k;
+}
+
+namespace {
+
+MatrixNmsKernelRef::DispatchData SetDefault(const matrix_nms_params& params, size_t idx) {
+    MatrixNmsKernelRef::DispatchData dispatch_data;
+
+    const auto& input_scores = params.inputs[1];
+    if (idx == 0) {
+        dispatch_data.gws = {input_scores.Batch().v, input_scores.Feature().v, 1};
+        dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+    } else if (idx == 1) {
+        dispatch_data.gws = {input_scores.Batch().v, 1, 1};
+        dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+    } else {
+        dispatch_data.gws = {1, 1, 1};
+        dispatch_data.lws = {1, 1, 1};
+    }
+
+    return dispatch_data;
+}
+}  // anonymous namespace
+
+KernelsData MatrixNmsKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
```

Comment:
is it possible to split the method to small logical methods with less than one screen long :)

Reply:
done